broker/src/test/java/io/camunda/zeebe/broker/jobstream/JobGatewayStreamerTest.java

```diff
@@ -0,0 +1,198 @@
+ */
+package io.camunda.zeebe.broker.jobstream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.atomix.cluster.MemberId;
+import io.atomix.cluster.messaging.ClusterCommunicationService;
+import io.atomix.cluster.messaging.ClusterEventService;
+import io.camunda.zeebe.broker.bootstrap.JobStreamServiceStep.DummyActivationProperties;
+import io.camunda.zeebe.broker.jobstream.ImmutableStreamRegistry.StreamConsumer;
+import io.camunda.zeebe.broker.jobstream.ImmutableStreamRegistry.StreamId;
+import io.camunda.zeebe.protocol.impl.record.value.job.JobRecord;
+import io.camunda.zeebe.protocol.impl.stream.JobStreamTopics;
+import io.camunda.zeebe.protocol.impl.stream.PushStreamRequest;
+import io.camunda.zeebe.scheduler.testing.ControlledActorSchedulerExtension;
+import io.camunda.zeebe.stream.api.ActivatedJob;
+import io.camunda.zeebe.stream.api.JobActivationProperties;
+import io.camunda.zeebe.util.buffer.BufferUtil;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.SortedSet;
+import java.util.TreeSet;
+import java.util.UUID;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.random.RandomGenerator;
+import org.agrona.CloseHelper;
+import org.agrona.DirectBuffer;
+import org.agrona.MutableDirectBuffer;
+import org.agrona.concurrent.UnsafeBuffer;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.mockito.Mockito;
+
+final class JobGatewayStreamerTest {
+  private final ClusterEventService eventService = Mockito.mock(ClusterEventService.class);
+  private final ClusterCommunicationService communicationService =
+      Mockito.mock(ClusterCommunicationService.class);
+  private final TestRegistry registry = new TestRegistry();
+  private final TestRandom random = new TestRandom();
+
+  private final JobGatewayStreamer streamer =
+      new JobGatewayStreamer(eventService, communicationService, registry, random);
+
+  @RegisterExtension
+  private final ControlledActorSchedulerExtension scheduler =
+      new ControlledActorSchedulerExtension();
+
+  @BeforeEach
+  void beforeEach() {
+    scheduler.submitActor(streamer);
+    scheduler.workUntilDone();
+  }
+
+  @AfterEach
+  void afterEach() {
+    CloseHelper.quietCloseAll(
+        () -> {
+          streamer.closeAsync();
+          scheduler.workUntilDone();
+        });
+  }
+
+  @Test
+  void shouldNotifyAvailableJobsWhenNoConsumers() {
+    // given
+    final var type = BufferUtil.wrapString("foo");
+
+    // when
+    final var maybeStream = streamer.streamFor(type);
+    scheduler.workUntilDone();
+
+    // then
+    assertThat(maybeStream).isEmpty();
+    Mockito.verify(eventService, Mockito.times(1))
+        .broadcast(JobStreamTopics.JOB_AVAILABLE.topic(), "foo");
+  }
+
+  @Test
+  void shouldPickRandomTarget() {
```

Comment:
Not super happy with this test, but I didn't have a better idea on how to test the random picking logic. Happy to remove it though, or extract the picking to its own class which will make this much easier to unit test.